topdown/topdown_test.go

```diff
@@ -3197,3 +3183,41 @@ func dump(note string, modules map[string]*ast.Module, data interface{}, docpath
 	}
 
 }
+
+func assertError(t *testing.T, expected interface{}, actual error) {
+	t.Helper()
+	if actual == nil {
+		t.Errorf("Expected error but got: %v", actual)
+		return
+	}
+
+	errString := actual.Error()
+
+	switch e := expected.(type) {
```

Comment:
It would be good to fail the test if the expected type does not match because otherwise we'll end up with silent failures.

Reply:
Done, and fixed a bunch of spots where we were asserting errors of different types in tests. This uncovered some inconsistencies with what various builtins are returning (and questions about what they should be returning), anyway.. filed https://github.com/open-policy-agent/opa/issues/2101 to track that.